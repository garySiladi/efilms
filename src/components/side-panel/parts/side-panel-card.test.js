// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import SidePanelCard from './side-panel-card';

const minimumCardDataProps = {href: 'xxx', icon: 'yyy', title: 'zzz'};

test('SidePanelCard renders with minimum props', () => {
  const component = renderer.create(<SidePanelCard {...minimumCardDataProps} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('SidePanelCard renders with isActive and onClick', () => {
  const component = renderer.create(
    <SidePanelCard
      {...minimumCardDataProps}
      isActive
      onClick={() => {}}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
