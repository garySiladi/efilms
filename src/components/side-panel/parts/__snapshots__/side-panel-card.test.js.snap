// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidePanelCard renders with isActive and onClick 1`] = `
<a
  className="side-panel-card"
  onClick={[Function]}
  style={Object {}}
>
  <span
    aria-hidden={true}
    className="fa fa-yyy side-panel-card__icon side-panel-card__icon--zzz"
  />
  <span
    className="side-panel-card__title"
  >
    zzz
  </span>
</a>
`;

exports[`SidePanelCard renders with minimum props 1`] = `
<a
  className="side-panel-card"
  onClick={[Function]}
  style={Object {}}
>
  <span
    aria-hidden={true}
    className="fa fa-yyy side-panel-card__icon side-panel-card__icon--zzz"
  />
  <span
    className="side-panel-card__title"
  >
    zzz
  </span>
</a>
`;
